dali/pipeline/executor/executor.h

```diff
@@ -679,11 +695,8 @@ void Executor<WorkspacePolicy, QueuePolicy>::PrepinData(
     for (int j = 0; j < node.spec.NumInput(); ++j) {
       auto tid = node.parent_tensors[j];
       // Use pinned memory only when it is useful
-      auto &parent_tensor_queue =
-          get_queue<OpType::CPU, StorageDevice::CPU>(tensor_to_store_queue_[tid]);
-      for (auto &tensor : parent_tensor_queue) {
-        tensor->set_pinned(node.spec.OutputDevice(0) == "gpu" && !RestrictPinnedMemUsage());
-      }
+      auto should_pin = node.spec.OutputDevice(0) == "gpu" && !RestrictPinnedMemUsage();
+      pin_cpu_passthrough(tensor_to_store_queue, graph, tid, should_pin);
```

Comment:
Do we want to ever propagate non-pinnedness here?
```suggestion
      if (node.spec.OutputDevice(0) == "gpu" && !RestrictPinnedMemUsage())
        pin_cpu_passthrough(tensor_to_store_queue, graph, tid);
```

Reply:
Not really, done.